tools/kompos/src/messages.ts

```diff
@@ -0,0 +1,207 @@
+
+export interface IdMap<T> {
+  [key: string]: T;
+}
+
+export interface Source {
+  id:         string;
+  sourceText: string;
+  mimeType:   string;
+  name:       string;
+  uri:        string;
+}
+
+export interface SimpleSourceSection {
+  firstIndex?: number;
+  length:      number;
+  line:        number;
+  column:      number;
+}
+
+export interface SourceSection extends SimpleSourceSection {
+  id:           string;
+  description?: string;
+  sourceId?:    string;
+}
+
+export interface Method {
+  name:          string;
+  definition:    SimpleSourceSection[];
+  sourceSection: SourceSection;
+}
+
+interface Frame {
+  sourceSection: SourceSection;
+  methodName: string;
+}
+
+interface TopFrame {
+  arguments: string[];
+  slots:     IdMap<string>;
+}
+
+export type Message = SourceMessage | SuspendEventMessage | MessageHistoryMessage;
```

Comment:
Yes, the idea is that there is only on set of type definitions for the tests and the web frontend.